STYLE_GUIDE.md

```diff
@@ -0,0 +1,156 @@
+ * Functions that return functions should have the suffix `Fn`
+ * A struct generally shouldn’t have a field named after itself, aka. this shouldn't occur:
+``` golang
+type middleware struct {
+	middleware Middleware
+}
+```
+ * In comments, use "iff" to mean, "if and only if".
+ * Product names are capitalized, like "Tendermint", "Basecoin", "Protobuf", etc except in command lines: `tendermint --help`
+ * Acronyms are all capitalized, like "RPC", "gRPC", "API".  "MyID", rather than "MyId".
+ * Prefer errors.New() instead of fmt.Errorf() unless you're actually using the format feature with arguments. (otherwise it's needlessly inefficient, & it'll crap out on %*)
+
+## Importing Libraries
+
+Sometimes it's necessary to rename libraries to avoid naming collisions or ambiguity.
+
+ * Use [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports)
+ * Separate imports into blocks - one for the standard lib, one for external libs and one for tendermint libs.
+ * Here are some common library labels for consistency:
+   - dbm "github.com/tendermint/tm-db"
+   - tmcmd "github.com/tendermint/tendermint/cmd/tendermint/commands"
+   - tmcfg "github.com/tendermint/tendermint/config/tendermint"
+   - tmtypes "github.com/tendermint/tendermint/types"
+ * Never use anonymous imports (the `.`), for example, `tmlibs/common` or anything else.
+ * When importing a pkg from the `tendermint/libs` directory, prefix the pkg alias with tm.
+     - tmbits "github.com/tendermint/tendermint/libs/bits"
+ * tip: Use the `_` library import to import a library for initialization effects (side effects)
+
+## Dependencies
+
+ * Dependencies should be pinned by a release tag, or specific commit, to avoid breaking `go get` when external dependencies are updated.
+
+## Testing
+
+ * The first rule of testing is: we add tests to our code
+ * The second rule of testing is: we add tests to our code
+ * For Golang testing:
+   * Make use of table driven testing where possible and not-cumbersome
+     - [Inspiration](https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go)
+   * Make use of [assert](https://godoc.org/github.com/stretchr/testify/assert) and [require](https://godoc.org/github.com/stretchr/testify/require)
+ * When using mocks, it is recommended to use either [testify mock](https://pkg.go.dev/github.com/stretchr/testify/mock
+ ) or [Mockery](https://github.com/vektra/mockery)
+
+## Errors
+
+ * Ensure that errors are concise, clear and traceable.
+ * Use stdlib errors package.
+ * For wrapping errors, use `fmt.Errorf()` with `%w`.
+ * Use panic sparingly only to expose internal problems.
+
+## Config
+
+ * Currently the TOML filetype is being used for config files
+ * A good practice is to store the default Config file under `~/.[yourAppName]/config.toml`
+
+## CLI
+
+ * When implementing a CLI use [Cobra](https://github.com/spf13/cobra) and [Viper](https://github.com/spf13/viper). urfave/cli can be replace with cobra in those repos where applicable.
```

Comment:
> urfave/cli can be replace with cobra in those repos where applicable.
What does this mean?  also lol at "urfave/cli"

Reply:
Not sure. I didn't write most of these parts. I just took them from one of the old tendermint repo's about style and just modified the top part plus take a quick skim through stuff that I'm sure wasn't applicable. My guess is that instead of using the `urfave/cli` package we use `cobra` instead but then this would make this sentence quite redundant

Reply:
I have just removed this